### PR TITLE
Refact NewError for reduce code lines.

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

## Description
refact NewError for reduce code lines.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: manager/apis/v2/handler/scheduler_cluster.go [hunk 1]
```diff
@@ -69,18 +64,14 @@ func (handler *Handler) DestroySchedulerCluster(ctx *gin.Context) {
...
-			ctx.JSON(http.StatusOK, "success")
-		} else {
-			NewError(ctx, http.StatusNotFound, errors.Newf("scheduler cluster not found, id %s", uri.ClusterID))
-		}
-	} else if dferrors.CheckError(err, dfcodes.InvalidResourceType) {
-		NewError(ctx, http.StatusBadRequest, err)
-	} else if dferrors.CheckError(err, dfcodes.ManagerStoreError) {
-		NewError(ctx, http.StatusInternalServerError, err)
+	if err != nil {
+		NewError(ctx, -1, err)
```
[the remark this answers is not in the document]
Comment (yxxhero) [reply manager/apis/v2/handler/scheduler_cluster.go:68]: I agree!  I will try do this.

[the remark this answers is not in the document]
Comment (yxxhero) [reply manager/apis/v2/handler/scheduler_cluster.go:68]: do you have some examples.  thanks very much.

[the remark this answers is not in the document]
Comment (yxxhero) [reply manager/apis/v2/handler/scheduler_cluster.go:68]: do you have some examples.  thanks very much.

File: manager/apis/v2/handler/scheduler_cluster.go [hunk 2]
```diff
@@ -28,25 +26,22 @@ import (
 func (handler *Handler) CreateSchedulerCluster(ctx *gin.Context) {
 	var cluster types.SchedulerCluster
 	if err := ctx.ShouldBindJSON(&cluster); err != nil {
-		NewError(ctx, http.StatusBadRequest, err)
+		ctx.Error(NewHTTPError(http.StatusBadRequest, err))
```
Comment (yxxhero) [inline manager/apis/v2/handler/scheduler_cluster.go:29]: @xujihui1985 like above?  Thanks very much.

File: manager/apis/v2/handler/scheduler_cluster.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (xujihui1985) [reply manager/apis/v2/handler/scheduler_cluster.go:68]: pretty much close, I think you could just throw the error, and handle the error in the global error handler looks like this one
```go
// ErrorHandle is a gin Middleware
func ErrorHandle() gin.HandlerFunc {
    return func (c *gin.Context) {
        c.Next()
        err := c.Errors.Last() // you can get all errors here
        if err == nil {
            return
        }

        // you can handle the error here, 
        if error, ok := errors.Cause(err.Err).(*myspace.KindOfClientError); ok {
            c.JSON(400, gin.H{
                "error": "Blah blahhh"
            })
            return 
        }
    }
```

**Review by gaius-qi — DISMISSED** [review]

File: manager/apis/v2/handler/cdn_cluster.go [hunk 3]
```diff
@@ -28,25 +26,21 @@ import (
 func (handler *Handler) CreateCDNCluster(ctx *gin.Context) {
 	var cluster types.CDNCluster
 	if err := ctx.ShouldBindJSON(&cluster); err != nil {
-		NewError(ctx, http.StatusBadRequest, err)
+		ctx.Error(NewError(http.StatusBadRequest, err))
```
Comment (yangjun289519474) [inline manager/apis/v2/handler/cdn_cluster.go:29]: 1.  ctx.Error just append a new error，NewError serializes the given struct as JSON into the response body and reply to http request.  they have different functions， and not equa.

File: manager/apis/v2/handler/error.go [hunk 4]
```diff
@@ -1,18 +1,52 @@
...
+	er := &HTTPError{
 		Message: err.Error(),
 	}
-	ctx.JSON(status, er)
+
+	if status > 0 {
+		er.Code = status
+		return er
+	}
+	e, ok := err.(*dferrors.DfError)
```
Comment (yangjun289519474) [inline manager/apis/v2/handler/error.go:34]: i dont't kown the benefit of the refactor. 
er.Code = int(e.Code) ,   it's not suitable to set internal error code as http error code

File: manager/apis/v2/handler/cdn_cluster.go (hunk 3 above)
Comment (yxxhero) [reply manager/apis/v2/handler/cdn_cluster.go:29]: @yangjun289519474 I  already change The NewError func. And I add a new handler.Error midware.

File: manager/apis/v2/handler/error.go (hunk 4 above)
Comment (yxxhero) [reply manager/apis/v2/handler/error.go:34]: @yangjun289519474 yes, you are right. I  will fix it. I am so sorry. I didn't pay attention to this.

**Comment by yxxhero** [comment]
Close because of https://github.com/dragonflyoss/Dragonfly2/pull/363

**Pull request closed without merging**